core/vm/evm.go

```diff
@@ -464,9 +513,16 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 		err = ErrMaxCodeSizeExceeded
 	}
 
-	// Reject code starting with 0xEF if EIP-3541 is enabled.
-	if err == nil && len(ret) >= 1 && ret[0] == 0xEF && evm.chainRules.IsLondon {
-		err = ErrInvalidCode
+	if err == nil && hasFormatByte(ret) {
+		if hasEIP3540(&evm.Config) {
+			// Allow only valid EOF1 if EIP-3540 is enabled.
+			if !validateEOF(ret) {
```

Comment:
When trying to deploy `EF` or `EFXX` where `XX != 00` it will fail with `ErrInvalidCodeFormat`. Maybe it's better to fail in this case with `ErrInvalidCode` as in London (because strictly speaking such codes are not EOF)